packages/ethereum/ethereum-light-client/src/consensus_state.rs

```diff
@@ -42,10 +42,68 @@ pub struct TrustedConsensusState {
     /// This sync committee can either be the current sync committee or the next sync
     /// committee. That's because the verifier uses next or current sync committee's
     /// public keys to verify the signature against.
-    pub sync_committee: ActiveSyncCommittee,
+    sync_committee: ActiveSyncCommittee,
 }
 
 impl TrustedConsensusState {
+    /// Creates a new trusted consensus state
+    /// # Errors
+    /// Returns an error if the untrusted sync committee does not match the trusted state
+    pub fn new<V: BlsVerify>(
+        trusted_state: ConsensusState,
+        untrusted_sync_committee: ActiveSyncCommittee,
+        bls_verifier: &V,
+    ) -> Result<Self, EthereumIBCError> {
+        match untrusted_sync_committee {
+            ActiveSyncCommittee::Current(ref committee) => {
+                ensure!(
+                    committee.aggregate_pubkey == trusted_state.current_sync_committee,
+                    EthereumIBCError::CurrenttSyncCommitteeMismatch {
+                        expected: trusted_state.current_sync_committee,
+                        found: committee.aggregate_pubkey
+                    }
+                );
+                let aggregate_pubkey = bls_verifier
```

Comment:
Since we do the same thing in both cases, can we just do that below once?

Reply:
done